### ISSUE TITLE
FunctionClauseError: no function clause matching in :pubkey_os_cacerts.conv_error_reason/1

### ISSUE THREAD

**Issue by mweidner037** [elixir-tesla/tesla#737]
I'm encountering the above error inside Tesla code after doing some updates - in particular, updating to Elixir 1.17 / OTP 27.

I believe it's related to a change in OTP 26: https://github.com/erlang/otp/issues/7321. I was wondering if anyone has advice for how to work around it, given that our app is not actually using Tesla directly - it's only a transitive dependency.

Here is the stack trace from our logging service:
```
FunctionClauseError: no function clause matching in :pubkey_os_cacerts.conv_error_reason/1

pubkey_os_cacerts.erl in :pubkey_os_cacerts.conv_error_reason/1 at line 266
    arg0	
    :no_cacerts_found
pubkey_os_cacerts.erl in :pubkey_os_cacerts.get/0 at line 45
httpc.erl in :httpc.ssl_verify_host_options/1 at line 806
httpc.erl in anonymous fn/0 in :httpc.http_options_default/0 at line 1416
httpc.erl in :httpc.http_options/3 at line 1362
httpc.erl in :httpc.handle_request/9 at line 1172
lib/tesla/adapter/httpc.ex in Tesla.Adapter.Httpc.request/2 at line 52
lib/tesla/adapter/httpc.ex in Tesla.Adapter.Httpc.call/2 at line 22
```

I've reproduced it with Tesla 1.8.0 and 1.13.2 on Elixir 1.17 / OTP 27. It does **not** occur with Tesla 1.8.0 on Elixir 1.15 / OTP 25.

**Comment by yordis** [comment]
It seems you are running a host without any OS certs; this is not even Tesla at that point but `:httpc.`

Are you running in a container? If so, you may be missing `apt-get install—y ca-certificates` or something similar.

What transitive deps is causing this? Most likely, you can inject your certs as well.

**Comment by mweidner037** [comment]
I added a stacktrace log to `deps/tesla/lib/tesla.ex#execute` and found that [oauth2](https://github.com/ueberauth/oauth2) was calling the default httpc adapter. Configuring it to use hackney instead fixes the error (`config :oauth2, adapter: Tesla.Adapter.Hackney`).

> Are you running in a container? If so, you may be missing `apt-get install -y ca-certificates` or something similar.

Yes, this is happening in `elixir:1.17.3-otp-27-slim`. I've added the ca-certificates install as a backup (I imagine this will recur if any other deps use httpc in the future).

Thank you!
